NIS module included in Debian image but not in Alpine one

Hello, the installed Python on Alpine and Debian is not strictly the same.
I'm trying to use the `nis` module on Alpine, but it is not compiled.

```
$ docker run python:jessie python -c 'import nis; print("OK")'
OK
$ docker run python:stretch python -c 'import nis; print("OK")'
OK
$ docker run python:slim python -c 'import nis; print("OK")'
OK
$ docker run python:alpine python -c 'import nis; print("OK")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'nis'
```

I read somewhere that the Alpine libc does not include `nis` built-in. But `libnsl` is available on Alpine repos (https://pkgs.alpinelinux.org/package/v3.7/community/x86_64/libnsl).
Can we imagine to have this feature in the future 3.7 Alpine image?

Thanks,
Antoine

I tried including the `libnsl-dev` package in #249 (it's only available on Alpine 3.7) but compiling the `nis` module still failed with:
```
building 'nis' extension
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I./Include -I. -I/usr/local/include -I/usr/src/python/Include -I/usr/src/python -c /usr/src/python/Modules/nismodule.c -o build/temp.linux-x86_64-3.6/usr/src/python/Modules/nismodule.o
/usr/src/python/Modules/nismodule.c:17:21: fatal error: rpc/rpc.h: No such file or directory
 #include <rpc/rpc.h>
                     ^
compilation terminated.
The following modules found by detect_modules() in setup.py, have been
built by the Makefile instead, as configured by the Setup files:
atexit                pwd                   time

Failed to build these modules:
nis
```

--- a/3.7-rc/alpine3.7/Dockerfile
+++ b/3.7-rc/alpine3.7/Dockerfile
@@ -46,6 +46,8 @@ RUN set -ex \
 		gdbm-dev \
 		libc-dev \
 		libffi-dev \
+		libnsl-dev \
+		libtirpc-dev \
 		linux-headers \
 		make \
 		ncurses-dev \

--- a/Dockerfile-alpine.template
+++ b/Dockerfile-alpine.template
@@ -40,6 +40,8 @@ RUN set -ex \
 		gdbm-dev \
 		libc-dev \
 		libffi-dev \
+		libnsl-dev \
+		libtirpc-dev \
 		linux-headers \
 		make \
 		ncurses-dev \

--- a/update.sh
+++ b/update.sh
@@ -158,6 +158,13 @@ for version in "${versions[@]}"; do
 			sed -ri -e 's/libressl/openssl/g' "$dir/Dockerfile"
 		fi
 
+		# Libraries to build the nis module available in Alpine 3.7, but also require this patch:
+		# https://bugs.python.org/issue32521
+		# TODO: Remove Python version check once 2.7 and 3.6 have the patch
+		if [[ "$variant" == alpine* ]] && [[ "$variant" != alpine3.7 || "$version" != 3.7* ]]; then
+			sed -ri -e '/libnsl-dev/d' -e '/libtirpc-dev/d' "$dir/Dockerfile"
+		fi
+
 		case "$v" in
 			wheezy/slim|jessie/slim)
 				sed -ri \

